### ISSUE TITLE
Create filtered permissions data component

### ISSUE THREAD

**Issue by JackHaeg** [hackforla/VRMS#1754]
### Overview
Currently, there is no way for admins to identify at scale which users are VRMS admins, or project managers (aka project leads) and which projects they are associated with on VRMS. Given that, we need to create a feature that enables admin users to filter & search for VRMS admins or all project leads linked to their respective projects, as well as view a full list of existing VRMS admins or project leads and their linked projects.

### Action Items
- [x] Create Filtered Permissions Data component on the “User Permission Search” screen (see #1752)
- [x] Takes props data and type
  - [x] `data` is an array of user data
  - [x] `type` is a string defining the type of data being passed
- [x] Render a MUI Text field 
  - [x] Filter data based on text field contents.
    - [x] Admin only filters by name string
    - [x] Project Leads filters by name string and by project string 
  - [x] Styled after the screenshots below:
<details><summary>Screenshot of text field with "Admin" selected</summary>
<p>

<img width="493" alt="Screenshot 2024-10-14 at 6 53 37 PM" src="https://github.com/user-attachments/assets/809d6d32-7c98-4daf-8227-94f3b3d0e0c5">

</p>
</details> 

<details><summary>Screenshot of text field with "Project Leads" selected</summary>
<p>

<img width="489" alt="Screenshot 2024-10-14 at 6 53 42 PM" src="https://github.com/user-attachments/assets/ff7dd2b2-db10-48dd-818e-7feae4a46bcf">

</p>
</details> 

- [x] Render data in a MUI Box component styled after the screenshots below 

<details><summary>Screenshot of admin selected</summary>
<p>

<img width="485" alt="Screenshot 2024-10-14 at 6 47 54 PM" src="https://github.com/user-attachments/assets/635a50e9-8566-498d-b456-e2d9af5c4dd6">

</p>
</details> 

<details><summary>Screenshot of Project Leads selected</summary>
<p>

<img width="487" alt="Screenshot 2024-10-14 at 6 48 04 PM" src="https://github.com/user-attachments/assets/d5a7d4fb-392a-4edd-991f-f8de84bdf9ed">

</p>
</details> 

- [x] Render user information based off of `type` 
  - [x] Each data element renders a link to the associated user profile.
  - [x] Admin = name and email
    - [x] Sort order: 
      - Alphabetically by first name 
  - [x] Project Leads = name and project 
    - [x] If a user is a project lead on multiple projects, list that user again for each project they are a part of. 
    - [x] Sort order:
      - alphabetically by project name first
      - alphabetically by first name within each project

### Resources/Instructions
<details><summary>Screenshot of User Permission Search Screen mockups</summary>
<p>
<img width="1248" alt="Screenshot 2024-10-07 at 8 30 51 PM" src="https://github.com/user-attachments/assets/1f1bc40c-b40f-4aa2-a8b7-9b34405991e1">

</p>
</details> 

- [Low-fi mockups created by Bonnie / Jack - see page 2 of mockups](https://docs.google.com/presentation/d/1oPP7mXDb09le3vVyaB4YUzOYYDa7sqDVCmvW5Srg30k/edit?usp=sharing)
- Use [MUI styling](https://mui.com/) on all components.
- This issue is part of this epic: #1737

**Comment by JackHaeg** [comment]
Hi @jng34 - This issue is part of a larger epic (https://github.com/hackforla/VRMS/issues/1737), which requires each issue to be completed in order. 

So, #1752 needs to be completed first before working on this issue. This issue will also rely on API calls that are still in development (see #1771). We will be un-assigning you from this issue for now and placing it in the icebox until these dependencies can be met. 

Please feel free to pick up #1752 in the meantime! And once all the dependencies are met, you can pick up this issue again :)

**Comment by jng34** [comment]
@JackHaeg Ahh, thank you for letting me know.

**Comment by vorleakyek** [comment]
@JackHaeg, @trillium, the search result in this ticket should be based on the dummy data created in https://github.com/hackforla/VRMS/issues/1752. Is that correct? It looks like we're doing the API integration in the next ticket #1801.

**Comment by JackHaeg** [comment]
@vorleakyek Yes, the search results in this ticket should be based on the dummy data created in #1752. The API integration occurs in #1801

**Comment by JackHaeg** [comment]
PR has been merged into [user_permissions_search branch](https://github.com/hackforla/VRMS/tree/user_permissions_search). Closing issue as completed.
